src/data/actions/assets.js

```diff
@@ -24,51 +16,120 @@ export const requestingAssets = () => ({
   type: assetActions.request.REQUESTING_ASSETS,
 });
 
-export const getAssets = (request, courseDetails) =>
+export const getAssets = (parameters, courseDetails) =>
   (dispatch, getState) => {
     dispatch(requestingAssets());
-    return clientApi.requestAssets(courseDetails.id, {
-      page: request.page,
-      assetTypes: request.assetTypes,
-      sort: request.sort,
-      direction: request.direction,
-    })
+
+    const state = getState();
+    const requestParameters = {
```

Comment:
Use the current state of the page to build request parameters to match, and then absorb the newly requested parameters.